src/opening.rs

```diff
@@ -91,6 +94,80 @@ where
 
         self.root == item
     }
+
+    /// Serialize an [`opening`] to a vector of bytes.
+    // Once the new implementation of the `Serializable` trait becomes
+    // available, we will want that instead, but for the time being we use
+    // this implementation.
+    pub fn to_var_bytes<const T_SIZE: usize>(&self) -> Vec<u8>
+    where
+        T: Serializable<T_SIZE>,
+    {
+        let mut bytes = Vec::new();
+
+        // serialize root
+        bytes.extend(&self.root.to_bytes());
+
+        // serialize branch
+        for level in self.branch.iter() {
+            for item in level.iter() {
+                bytes.extend(&item.to_bytes());
+            }
+        }
+
+        // serialize positions
+        for pos in self.positions.iter() {
+            // the positions will be in the range [0..A[, so casting to u32
+            // is never going to be a problem
+            #[allow(clippy::cast_possible_truncation)]
+            bytes.extend(&(*pos as u32).to_bytes());
+        }
+
+        bytes
+    }
+
+    /// Deserialize an [`Opening`] from a slice of bytes.
+    ///
+    /// # Errors
+    ///
+    /// Will return [`dusk_bytes::Error`] in case of a deserialization error.
+    // Once the new implementation of the `Serializable` trait becomes
+    // available, we will want that instead, but for the time being we use
+    // this implementation.
+    pub fn from_slice<const T_SIZE: usize>(
+        buf: &[u8],
+    ) -> Result<Self, BytesError>
+    where
+        T: Serializable<T_SIZE>,
+        <T as Serializable<T_SIZE>>::Error: dusk_bytes::BadLength,
+        dusk_bytes::Error: From<<T as Serializable<T_SIZE>>::Error>,
+    {
+        let mut bytes = buf;
```

Comment:
Since we know the size here we could also return a `BytesError::BadLength` if the size is not what we expect.

Reply:
This is a really good suggestion as it actually makes the test fail :sweat_smile: 
Will investigate why the lengths don't match

Reply:
My bad, the positions are encoded as `u32` (`Serializable` is not implemented for `usize`), so the expected length was calculated incorrectly. 
Everything should work now!